api/server/router/system/system_routes.go

```diff
@@ -43,6 +47,24 @@ func (s *systemRouter) pingHandler(ctx context.Context, w http.ResponseWriter, r
 	return err
 }
 
+func (s *systemRouter) swarmState() string {
+	if s.cluster == nil {
+		return string(swarm.LocalNodeStateInactive)
+	}
+
+	swarmState := string(swarm.LocalNodeStateInactive)
+	ci := s.cluster.Info()
+	swarmState = string(ci.LocalNodeState)
+	if ci.LocalNodeState == swarm.LocalNodeStateActive {
+		if ci.ControlAvailable {
```

Comment:
I think `ControlAvailable` means it is the current leader, not if it is a manager (outside of being the leader).

Reply:
Hm.. you may be right; https://github.com/moby/moby/blob/46cdcd206c56172b95ba5c77b827a722dab426c5/daemon/cluster/noderunner.go#L388-L396

Reply:
Oh! That's slightly different, but I think `controlClient` would match that. Let me dig a bit further

Reply:
Right, so this is the `cluster.Info()` (which is used to propagate the `docker info` data); https://github.com/moby/moby/blob/46cdcd206c56172b95ba5c77b827a722dab426c5/daemon/cluster/swarm.go#L447-L454

And looks like that's the field that's used by the CLI to print if the node is a Manager; https://github.com/docker/cli/blob/88c6089300a82d3373892adf6845a4fed1a4ba8d/cli/command/system/info.go#L311-L319

```go
func printSwarmInfo(dockerCli command.Cli, info types.Info) {
	if info.Swarm.LocalNodeState == swarm.LocalNodeStateInactive || info.Swarm.LocalNodeState == swarm.LocalNodeStateLocked {
		return
	}
	fmt.Fprintln(dockerCli.Out(), "  NodeID:", info.Swarm.NodeID)
	if info.Swarm.Error != "" {
		fmt.Fprintln(dockerCli.Out(), "  Error:", info.Swarm.Error)
	}
	fmt.Fprintln(dockerCli.Out(), "  Is Manager:", info.Swarm.ControlAvailable)
```